Parameterize tests for Answer interface

jprado1 opened the pull request:
I use the `parameterized` library to parameterise tests for the Answer interface with a @parameterized_class decorator:

`@parameterized_class(("name","tested_answer"), [
   ("TextareaAnswer",TextareaAnswer(), ),
   ("CodeDemo",CodeDemo(
            code_input=  WidgetCodeInput(
                function_name="name", 
                function_parameters="",
                docstring="""""",
                function_body="""body"""),
            input_parameters_box=None,
            visualizers=None,
            update_visualizers=None,
            code_checker=None,
            separate_check_and_update_buttons=False,
            update_on_input_parameter_change=False
), )
])`

creates two tests named `TestAnswerRegistry_{testnumber}_{name}`. parameterized_class is used in order to use parameters in setUp and eventually tearDown. To add new parameterized tests, it suffices to add tuples (name : String, answer : Answer) to the list defined inside the decorator.

agoscinski commented on python/tests/tests.py:
```diff
@@ -143,14 +147,33 @@ def erroneous_update_function(a, wci, visualizers):
+        self.answer_registry.register_answer_widget("textarea_key", self.answer)
         self.answer_registry._output = SurpressStdOutput()
         InteractiveShell.instance()
 
```
We just need a `tearDown` function which deletes the `test-MaxMustermann` json file if it exists after each tests. This is important, because when multiple tests will access the json file, the test can rely on that they created the `test-MaxMustermann.json`file and they don't read the from a file created for a different test.

jprado1 commented:
# 1, Fixed typo in SupressStdOutput class name
# 2. Corrected test for update() and created test check() when there is no button
# 3. removed old TODOs and general clean up
# 4. Adapted AnswerTest with respect to fix of PR #7 (but did not implement the respective test of this PR)

agoscinski commented on python/examples/visual_cues_examples.ipynb:
```diff
@@ -0,0 +1,452 @@
+    "\n",
+    "\n",
+    "\n",
+    "from ipycanvas import Canvas\n",
```
Can you add one cell before that with
```!pip install ipycanvas```
so tox can also run all notebooks

jprado1 replied:
Done, will merge

agoscinski reviewed (changes requested):
Looks good, just one minor change and it can be merged
